File: core/core-domain/src/main/java/com/nmh/commerce/product/Product.java
package com.nmh.commerce.product;

import com.nmh.commerce.domain.Money;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Product {
    private final Long id;
    private final String name;
    private final Money price;

    @Builder
    private Product(Long id, String name, Money price) {
        this.id = id;
        this.name = name;
        this.price = price;
    }
}

File: core/core-domain/src/main/kotlin/com/nmh/commerce/product/Product.kt
package com.nmh.commerce.product

import com.nmh.commerce.domain.Money

data class Product(
    val id: Long,
    val name: String,
    val price: Money,
) {
    companion object {
        fun of(
            id: Long,
            name: String,
            price: Money,
        ): Product = Product(id, name, price)
    }
}
